nvflare/job_config/script_runner.py

```diff
@@ -270,43 +251,85 @@ def add_to_fed_job(self, job: FedJob, ctx, **kwargs):
             executor = (
                 self._executor
                 if self._executor
-                else self._get_in_process_executor_cls(self._framework)(
+                else InProcessClientAPIExecutor(
                     task_script_path=self._script,
                     task_script_args=self._script_args,
-                    params_exchange_format=self._params_exchange_format,
+                    params_exchange_format=self._client_script_params_exchange_format,
                     params_transfer_type=self._params_transfer_type,
-                    from_nvflare_converter_id=self._from_nvflare_converter_id,
-                    to_nvflare_converter_id=self._to_nvflare_converter_id,
                 )
             )
             job.add_executor(executor, tasks=tasks, ctx=ctx)
 
+        self._add_params_converter_filter(job=job, ctx=ctx)
         job.add_resources(resources=[self._script], ctx=ctx)
         return comp_ids
 
-    def _get_ex_process_executor_cls(self, framework: FrameworkType) -> Type[ClientAPILauncherExecutor]:
-        if framework == FrameworkType.PYTORCH:
-            from nvflare.app_opt.pt.client_api_launcher_executor import PTClientAPILauncherExecutor
-
-            return PTClientAPILauncherExecutor
-        elif framework == FrameworkType.TENSORFLOW:
-            from nvflare.app_opt.tf.client_api_launcher_executor import TFClientAPILauncherExecutor
-
-            return TFClientAPILauncherExecutor
-        else:
-            return ClientAPILauncherExecutor
+    def _add_params_converter_filter(self, job, ctx):
+        if (
```

Comment:
Think about better re-organizing this block code.
You can use _SUPPORT_EXCHANGE_FORMAT to register a "add_filters_cb" for each supported combination. Then this chunk of code simply needs to call the CB of the combo.

Reply:
changed as suggested for these pre-built scenario.